### PR TITLE
Create planning extension

### PR DESCRIPTION
**Pull request by tomaskikutis**
DEPENDS ON: https://github.com/superdesk/superdesk-client-core/pull/2966

### CODE REVIEW

File: scripts/apps/archive/index.tsx [hunk 1]
```diff
@@ -406,17 +417,83 @@ function spikeActivity(spike, data, modal, $location, $q, multi, privileges,
...
         }
 
-        return modal.confirm(message, gettext('Confirm'))
-            .then(() => spike.spike(data.item));
+        const onSpikeMiddlewares
+            : Array<IExtensionActivationResult['contributions']['entities']['article']['onSpike']>
+            = flatMap(
+                Object.values(extensions).map(({activationResult}) => activationResult),
+                (activationResult) =>
+                    activationResult.contributions != null
```
Comment (pablopunk) [inline scripts/apps/archive/index.tsx]: Javascript needs optional chaining ASAP... It's on stage 2 right now https://github.com/tc39/proposal-optional-chaining 🤞🏻

Comment (pablopunk) [inline scripts/apps/archive/index.tsx]: Javascript needs optional chaining ASAP... It's on stage 2 right now https://github.com/tc39/proposal-optional-chaining 🤞🏻

File: scripts/apps/search/controllers/MultiActionBarController.tsx [hunk 2]
```diff
@@ -115,27 +124,91 @@ export function MultiActionBarController(
...
-            message = gettext('Some item/s are linked to in-progress planning coverage, spike anyway?');
-        } else if (!_.get(config, 'confirm_spike', true)) {
-            spikeMultiple();
+            modal.confirm(gettext('Do you want to delete the items permanently?')).then(spikeMultiple);
             return;
         }
 
-        return modal.confirm(message)
-            .then(spikeMultiple);
+        const onSpikeMultipleMiddlewares
```
Comment (pablopunk) [inline scripts/apps/search/controllers/MultiActionBarController.tsx]: From this line forwards, isn't it all duplicated code from `scripts/apps/archive/index.tsx`?

Comment (tomaskikutis) [reply scripts/apps/search/controllers/MultiActionBarController.tsx]: [not duplicate](https://user-images.githubusercontent.com/2076953/59419505-5b3b4800-8dcb-11e9-9f33-85a4d3c86a91.png), but similar. I've moved it to a separate function.

**Pull request merged**